### ISSUE TITLE
[功能需求]DataCollatorForLanguageModeling（动态masking）

### ISSUE THREAD

**Issue by tanhanzhuo** [PaddlePaddle/PaddleNLP#1340]
大数据训练模型时，huggingface的transformers预训练采取的是DataCollatorForLanguageModeling进行动态masking（训练过程中mask数据）。目前examples/language_model/bert/采取的是静态masking（提前计算masking然后储存），examples/language_model/ernie-1.0/采取的动态masking，但代码调用的函数和库很复杂，不容易定制。是否可以增加huggingface对应的DataCollatorForLanguageModeling系列函数（比如DataCollatorForWholeWordMask对应ernie1.0的masking机制）？

DataCollatorForLanguageModeling will pad, mask, add_special_tokens, and etc. to the input data during training (dynamic masking)
link to the "DataCollatorForLanguageModeling":
https://huggingface.co/transformers/main_classes/data_collator.html#transformers.data.data_collator.DataCollatorForLanguageModeling

sample usage:
https://github.com/huggingface/transformers/blob/efea0f868bd381244e3cef51b388293e41a36d1e/examples/pytorch/language-modeling/run_mlm_no_trainer.py#L447

from transformers import DataCollatorForLanguageModeling
data_collator = DataCollatorForLanguageModeling(tokenizer=tokenizer, mlm_probability=args.mlm_probability)
train_dataloader = DataLoader(
    train_dataset, shuffle=True, collate_fn=data_collator, batch_size=args.per_device_train_batch_size
)
for step, batch in enumerate(train_dataloader):
    # start the training

实际上paddle的finetuning文档采取的是这种DataLoader模式处理数据，
https://github.com/PaddlePaddle/PaddleNLP/blob/7c78894b8f79d531ee8790a81cfa86a937fcd38b/examples/benchmark/glue/run_glue.py#L266
增加预训练的collate_fn可以很好的让预训练框架简化。
